core/templates/pages/community-dashboard-page/services/contribution-opportunities.service.ts

```diff
@@ -17,73 +17,108 @@
  * fields.
  */
 
-require(
-  'pages/community-dashboard-page/services/' +
-  'contribution-opportunities-backend-api.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('ContributionOpportunitiesService', [
-  'ContributionOpportunitiesBackendApiService',
-  function(ContributionOpportunitiesBackendApiService) {
-    var skillOpportunitiesCursor = null;
-    var translationOpportunitiesCursor = null;
-    var voiceoverOpportunitiesCursor = null;
-    var moreSkillOpportunitiesAvailable = true;
-    var moreTranslationOpportunitiesAvailable = true;
-    var moreVoiceoverOpportunitiesAvailable = true;
+import { ContributionOpportunitiesBackendApiService,
+  IFetchOpportunitySuccessCallbackParams }
+// eslint-disable-next-line max-len
+  from 'pages/community-dashboard-page/services/contribution-opportunities-backend-api.service';
+import { SkillOpportunity } from
+  'domain/opportunity/SkillOpportunityObjectFactory';
 
-    var _getSkillOpportunities = function(cursor, successCallback) {
-      ContributionOpportunitiesBackendApiService.fetchSkillOpportunities(
-        cursor).then((opportunities, nextCursor, more) => {
-        skillOpportunitiesCursor = nextCursor;
-        moreSkillOpportunitiesAvailable = more;
-        successCallback(opportunities, more);
-      });
-    };
-    var _getTranslationOpportunities = function(
-        languageCode, cursor, successCallback) {
-      ContributionOpportunitiesBackendApiService.fetchTranslationOpportunities(
-        languageCode, cursor).then((opportunities, nextCursor, more) => {
-        translationOpportunitiesCursor = nextCursor;
-        moreTranslationOpportunitiesAvailable = more;
-        successCallback(opportunities, more);
+@Injectable({
+  providedIn: 'root'
+})
+export class ContributionOpportunitiesService {
+  skillOpportunitiesCursor = null;
+  translationOpportunitiesCursor = null;
+  voiceoverOpportunitiesCursor = null;
+  moreSkillOpportunitiesAvailable = true;
+  moreTranslationOpportunitiesAvailable = true;
+  moreVoiceoverOpportunitiesAvailable = true;
+
+  constructor(
+    private contributionOpportunitiesBackendApiService:
+      ContributionOpportunitiesBackendApiService) { }
+
+  private _getSkillOpportunities(cursor: string,
+      successCallback: (opportunities: SkillOpportunity[],
+      more: boolean) => void): void {
+    this.contributionOpportunitiesBackendApiService.fetchSkillOpportunities(
+      cursor).then((value: IFetchOpportunitySuccessCallbackParams) => {
```

Comment:
Why can't we have the three values separately? (Any specific reason for this change? I see we are passing values separately on line 52, why not here?)

Reply:
The linter is not allowing me to pass 3 values according to this image:
<img width="850" alt="Screenshot 2020-05-03 at 19 11 16" src="https://user-images.githubusercontent.com/11369764/81180873-ec9b4600-8fab-11ea-8aee-66782f9bf04c.png">

Reply:
**UPDATE**

@DubeySandeep refer to my comment: https://github.com/oppia/oppia/pull/9230#discussion_r439045478